qutip/rcsolve.py

```diff
@@ -0,0 +1,152 @@
+# This file is part of QuTiP: Quantum Toolbox in Python.
+#
+#    Copyright (c) 2011 and later, Paul D. Nation and Robert J. Johansson.
+#    All rights reserved.
+#
+#    Redistribution and use in source and binary forms, with or without
+#    modification, are permitted provided that the following conditions are
+#    met:
+#
+#    1. Redistributions of source code must retain the above copyright notice,
+#       this list of conditions and the following disclaimer.
+#
+#    2. Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#
+#    3. Neither the name of the QuTiP: Quantum Toolbox in Python nor the names
+#       of its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+#    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+#    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#    HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+#Author: Neill Lambert, Anubhav Vardhan
+#Contact: nwlambert@gmail.com
+
+import time
+import warnings
+import numpy as np
+import scipy.sparse as sp
+from numpy import matrix
+from numpy import linalg
+from qutip import spre, spost, sprepost, thermal_dm, mesolve, Odeoptions
+from qutip import tensor, identity, destroy, sigmax, sigmaz, basis, qeye, dims
+
+def rcsolve(Hsys, Q, wc, alpha, N, Temperature, tlist, initial_state,
+            return_vals, eigen_sparse=False, options=None):
+    """
+    Function to solve for an open quantum system using the
+    reaction coordinate (RC) model. 
+
+    Parameters
+    ----------
+    Hsys: Qobj
+        The system hamiltonian.
+    Q: Qobj
+        The coupling between system and bath.
+    wc: Float
+        Cutoff frequency.
+    alpha: Float
+        Coupling strength.
+    N: Integer
+        Number of cavity fock states.
+    Temperature: Float
+        Temperature. 
+    tlist: List.
+        Time over which system evolves.
+    initial_state: Qobj
+        Initial state of the system.
+    return_vals: list of :class:`qutip.Qobj` / callback function single
```

Comment:
usually we call this argument `e_ops`. any particular reason to not follow that convention?